org.eclipse.scanning.example/src/org/eclipse/scanning/example/malcolm/DummyMalcolmRecord.java

```diff
@@ -0,0 +1,453 @@
+        if (!underControl) {
+            underControl = true;
+            return true;
+        }
+        return false;
+    }
+
+    synchronized void releaseControl() {
+        underControl = false;
+    }
+
+    private class RPCServiceAsyncImpl implements RPCServiceAsync {
+
+        private DummyMalcolmRecord pvRecord;
+        private final Status statusOk = StatusFactory.
+                getStatusCreate().getStatusOK();
+        private String methodName = "";
+
+        RPCServiceAsyncImpl(DummyMalcolmRecord record, String methodName) {
+            pvRecord = record;
+            this.methodName = methodName;
+        }
+
+        public void request(PVStructure args, RPCResponseCallback callback)
+        {
+        	System.out.println("Got Async Request:");
+        	System.out.println(args.toString());
+        	receivedRPCCalls.put(methodName, args);
+        	
+            boolean haveControl = pvRecord.takeControl();
+            if (!haveControl)
+            {
+                handleError("Device busy", callback, haveControl);
+                return;
+            }
+            
+
+            Structure mapStructure = fieldCreate.createFieldBuilder().
+        			setId("malcolm:core/Map:1.0").
+        			createStructure();
+            PVStructure returnPvStructure = pvDataCreate.createPVStructure(mapStructure);
+            
+            if ("validate".equals(methodName)) {
+            	returnPvStructure = args;
+            } else if ("configure".equals(methodName)) {
+            	pvRecord.getPVStructure().getSubField(PVString.class, "state.value").put("CONFIGURING");
+            } else if ("run".equals(methodName)) {
+                pvRecord.getPVStructure().getSubField(PVString.class, "state.value").put("RUNNING");
+                try {
+    				Thread.sleep(2000);
+    			} catch (InterruptedException e1) {
+    				e1.printStackTrace();
+    			}
+            }
+
+
+        	pvRecord.getPVStructure().getSubField(PVString.class, "state.value").put("READY");
+            
+        	pvRecord.releaseControl();
+            callback.requestDone(statusOk, returnPvStructure);
+            return;
+            
+        }
+
+        private void handleError(String message, RPCResponseCallback callback, boolean haveControl)
+        {
+            if (haveControl)
+                pvRecord.releaseControl();
+            Status status = StatusFactory.getStatusCreate().
+                    createStatus(StatusType.ERROR, message, null);
+            callback.requestDone(status, null);
+        }
+    }
+
+    public static DummyMalcolmRecord create(String recordName)
+    {
+        FieldBuilder fb = fieldCreate.createFieldBuilder();
+        
+
+        Structure metaStructure = fieldCreate.createFieldBuilder().
+    			add("description", ScalarType.pvString).
+    			addArray("tags", ScalarType.pvString).
+    			add("writeable", ScalarType.pvBoolean).
+    			add("label", ScalarType.pvString).

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "label" 12 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)